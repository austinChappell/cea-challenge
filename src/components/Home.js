import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';

class Home extends Component {
  componentWillMount() {
    console.log('PROPS', this.props.match);
  }

  render() {
    const { accessToken } = this.props;
    const meetupAuthUrl = `https://secure.meetup.com/oauth2/authorize?client_id=${process.env.REACT_APP_MEETUP_CONSUMER_KEY}&response_type=token&redirect_uri=${process.env.REACT_APP_MEETUP_REDIRECT_URI}`;

    const link = accessToken ?
      <Redirect to="/event_details" />
      : <a href={meetupAuthUrl}>Enter ReactJS Dallas</a>;

    return (
      <div className="Home">
        Home Component
        {link}
      </div>
    );
  }
}

export default Home;
