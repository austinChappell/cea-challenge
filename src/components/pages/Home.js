import React, { Component } from 'react';
import { Link, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

const myMeetups = [
  {
    title: 'React',
    url: 'ReactJS-Dallas',
  },
  {
    title: 'NodeSchool',
    url: 'Nodeschool-Dallas',
  },
];

class Home extends Component {

  state = {
    redirect: false,
  }

  componentWillMount() {
    if (!this.props.accessToken) {
      const { hash } = this.props.location;
      if (hash.indexOf('access_token=') > -1) {
        // split the url query string on "access_token="
        const splitHash = hash.split('access_token=');

        // split the other query params off and return only the token
        const accessToken = splitHash[1].split('&')[0];

        // save accessToken to Redux
        this.props.setAccessToken(accessToken);
        this.setRedirect();
      }
    }
  }

  componentWillUpdate(nextProps) {
    if (this.props.accessToken !== nextProps.accessToken) {
      this.setRedirect();
    }
  }

  setRedirect = () => {
    this.setState({ redirect: true })
  }

  storeGroupUrl = (url) => {
    console.log('URL', url)
    sessionStorage.setItem('groupUrl', url);
  }

  render() {
    const { accessToken } = this.props;
    const meetupAuthUrl = `https://secure.meetup.com/oauth2/authorize?client_id=${process.env.REACT_APP_MEETUP_CONSUMER_KEY}&response_type=token&redirect_uri=${process.env.REACT_APP_MEETUP_REDIRECT_URI}`;
    const groupUrl = sessionStorage.getItem('groupUrl');
    console.log('REDIRECT', this.state.redirect)

    const link = accessToken && this.state.redirect ?
      <Redirect to={`/group/${groupUrl}`} />
      : <a href={meetupAuthUrl}>Enter ReactJS Dallas</a>;

    return (
      <div className="Home page">
        Home Component
        {link}
        {myMeetups.map((meetup, index) => {
          const redirectLink = accessToken ? 
            <Link to={`/group/${meetup.url}`} />
            : 
            <a href={meetupAuthUrl} onClick={() => this.storeGroupUrl(meetup.url)}>{meetup.title}</a>;
          const meetupLink = this.state.redirect ?
            <Redirect to={`/group/${sessionStorage.groupUrl}`} />
            : redirectLink;

          return (
            <div key={index}>
              {meetupLink}
            </div>
          );
        })}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  accessToken: state.generalReducer.accessToken,
});

const mapDispatchToProps = dispatch => ({
  setAccessToken: (accessToken) => {
    const action = { type: 'SET_ACCESS_TOKEN', accessToken };
    dispatch(action);
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(Home);
