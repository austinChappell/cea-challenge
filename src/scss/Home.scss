.Home {
  // background: linear-gradient($color-black, $color-secondary-light);
  background: url('./assets/images/kari-shea-226153-unsplash.jpg');
  background-size: cover;
  background-attachment: fixed;
  text-align: center;
  position: relative;
  .overlay {
    position: absolute;
    overflow: auto;
    height: 100%;
    background-color: rgba(10, 0, 10, 0.8);
  }
  h1 {
    color: $color-secondary;
    margin: 20vh 20px 20px;
  }
  .meetups {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    width: 80%;
    margin: 0 auto;
  }
  .meetup {
    background-color: $color-primary;
    margin: 30px;
    display: grid;
    justify-content: center;
    align-content: center;
  }
  a {
    color: $color-white;
    text-decoration: none;
    font-size: 2rem;
    padding: 10px 25px;
  }
}