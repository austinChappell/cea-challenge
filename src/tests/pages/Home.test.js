import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import Home from '../../components/pages/Home';

const initialState = {
  generalReducer: {
    accessToken: 'fakeToken',
  },
};

const middlewares = [];
const mockStore = configureStore(middlewares);

describe('The Home Component', () => {
  it('should render correctly', () => {
    const options = {
      context: {
        store: mockStore(initialState),
      },
    };
    const wrapper = shallow(<Home />, options);
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
