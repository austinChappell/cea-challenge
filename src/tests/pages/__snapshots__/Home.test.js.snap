// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Home Component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    accessToken="fakeToken"
    setAccessToken={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Austin's Favorite Meetups
        </h1>,
        Array [
          <div>
            <Link
              replace={false}
              to="/group/ReactJS-Dallas"
            >
              React
            </Link>
          </div>,
          <div>
            <Link
              replace={false}
              to="/group/Nodeschool-Dallas"
            >
              NodeSchool
            </Link>
          </div>,
        ],
      ],
      "className": "Home page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Austin's Favorite Meetups",
        },
        "ref": null,
        "rendered": "Austin's Favorite Meetups",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/group/ReactJS-Dallas"
          >
            React
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "React",
            "replace": false,
            "to": "/group/ReactJS-Dallas",
          },
          "ref": null,
          "rendered": "React",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/group/Nodeschool-Dallas"
          >
            NodeSchool
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "NodeSchool",
            "replace": false,
            "to": "/group/Nodeschool-Dallas",
          },
          "ref": null,
          "rendered": "NodeSchool",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Austin's Favorite Meetups
          </h1>,
          Array [
            <div>
              <Link
                replace={false}
                to="/group/ReactJS-Dallas"
              >
                React
              </Link>
            </div>,
            <div>
              <Link
                replace={false}
                to="/group/Nodeschool-Dallas"
              >
                NodeSchool
              </Link>
            </div>,
          ],
        ],
        "className": "Home page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Austin's Favorite Meetups",
          },
          "ref": null,
          "rendered": "Austin's Favorite Meetups",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/group/ReactJS-Dallas"
            >
              React
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "React",
              "replace": false,
              "to": "/group/ReactJS-Dallas",
            },
            "ref": null,
            "rendered": "React",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/group/Nodeschool-Dallas"
            >
              NodeSchool
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "NodeSchool",
              "replace": false,
              "to": "/group/Nodeschool-Dallas",
            },
            "ref": null,
            "rendered": "NodeSchool",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
